synapse/models/inet.py

```diff
@@ -3306,6 +3464,267 @@ def getModelDefs(self):
+                    )),
+
+                    ('inet:service:message:link', {}, (
+
+                        ('title', ('str', {}), {
+                            'doc': 'The title text for the link.'}),
+
+                        ('url', ('inet:url', {}), {
+                            'doc': 'The URL which was attached to the message.'}),
+                    )),
+
+                    ('inet:service:message:attachment', {}, (
+
+                        ('name', ('file:path', {}), {
+                            'doc': 'The name of the attached file.'}),
+
+                        ('text', ('str', {}), {
+                            'doc': 'Any text associated with the file such as alt-text for images.'}),
+
+                        ('file', ('file:bytes', {}), {
+                            'doc': 'The file which was attached to the message.'}),
+                    )),
+
+                    ('inet:service:channel', {}, (
+
+                        ('name', ('str', {'onespace': True, 'lower': True}), {
+                            'doc': 'The current name of the channel.'}),
+
+                        ('period', ('ival', {}), {
+                            'doc': 'The time period where the channel was available.'}),
+                    )),
+
+                    ('inet:service:channel:member', {}, (
+
+                        ('channel', ('inet:service:channel', {}), {
+                            'doc': 'The channel that the account was a member of.'}),
+
+                        ('account', ('inet:service:account', {}), {
+                            'doc': 'The account that was a member of the channel.'}),
+
+                        ('period', ('ival', {}), {
+                            'doc': 'The time period where the account was a member of the channel.'}),
+                    )),
+
+                    ('inet:service:resource:type:taxonomy', {}, {}),
+                    ('inet:service:resource', {}, (
+
+                        ('name', ('str', {'onespace': True, 'lower': True}), {
+                            'doc': 'The name of the service resource.'}),
+
+                        ('desc', ('str', {}), {
+                            'disp': {'hint': 'text'},
+                            'doc': 'A description of the service resource.'}),
+
+                        ('url', ('inet:url', {}), {
+                            'doc': 'The primary URL where the resource is available from the service.'}),
+
+                        ('type', ('inet:service:resource:type:taxonomy', {}), {
+                            'doc': 'The resource type. For example "s3.bucket", "file", or "rpc.endpoint".'}),
+
+                        ('file', ('file:bytes', {}), {
+                            'doc': 'The file bytes if this resource represents a static file such as an S3 bucket or file share.'}),
+                    )),
+
+                    ('inet:service:access', {}, (
+
+                        ('resource', ('inet:service:resource', {}), {
```

Comment:
Should this include a reference to a inet:service:rule, inet:service:permission, or <some type of credentials> to indicate the type of access attempted? Leave as TODO for now?

Reply:
Would the referenced rule essentially mean "this is the rule which allowed/denied the access" ?

Reply:
Hrrm. Was actually thinking of "type of access requested".  If access fails, an error message (if available) might let me know what someone tried to do ("thesilence attempted to delete visi's macro"). But in the case of "access succeeded" (or the absence of an error message) , the form doesn't seem to capture what the requestor  did / tried to do.

"Type of access requested" is not necessarily 1:1 with "rule that allowed or denied access". :thinking: Do we think it's a close enough proxy? Am I down a rabbit hole?